interfaces/thirdparty/serenity.py

```diff
@@ -0,0 +1,182 @@
+from scm.plams.core.basejob import SingleJob
+from scm.plams.core.results import Results
+from scm.plams.core.settings import Settings
+from scm.plams.mol.molecule import Molecule
+from scm.plams.tools.units import Units
+
+from pathlib import Path
+import numpy as np
+import h5py
+import shutil
+import os
+
+__all__ = ["SerenityJob", "SerenityResults", "SerenitySettings"]
+
+
+class SerenitySettings(Settings):
+    """A Subclass of the Settings class to set the iteration order through keys."""
+
+    def __iter__(self):
+        """Iteration through keys follows insertion order instead of lexicographical order."""
+        return iter(self.keys())
+
+
+class SerenityResults(Results):
+    """A Subclass of the Results class to access the results given in the Serenity output. Can be extended as required."""
+
+    def _get_energy_type(self, search="Total Energy", index=-1, unit="a.u."):
+        s = self.grep_output(search)
+        s = [item for item in s if not "..." in item]
+        s = s[index]
+
+        def convert_with_fallback(item):
+            try:
+                return float(item.split()[-1])
+            except ValueError:
+                try:
+                    return float(item.split()[-2])
+                except ValueError:
+                    raise ValueError("Can not convert type to float.")
+
+        if not isinstance(index, slice):
+            return Units.convert(convert_with_fallback(s), "a.u.", unit)
+        else:
+            return [Units.convert(convert_with_fallback(x), "a.u.", unit) for x in s]
+
+    def get_energy(self, index=-1, unit="a.u."):
+        """Returns 'Total Energy (DFT)/(HF):' from the output file. Set ``index`` to choose the n-th occurence of the total energy in the output. Defaults to the last occurence."""
+        return self._get_energy_type("Total Energy", index=index, unit=unit)
+
+    def get_ccsd_energy_correction(self, index=-1, unit="a.u."):
+        """Returns the 'CCSD Energy Correction' from the output file. Set ``index`` to choose the n-th occurrence of the CCSD energy correction in the output. Defaults to the last occurrence."""
+        return self._get_energy_type("CCSD Energy Correction", index=index, unit=unit)
+
+    def get_triples_energy_correction(self, index=-1, unit="a.u."):
+        return self._get_energy_type("Triples Energy Correction", index=index, unit=unit)
+
+    def get_local_ccsd_energy(self, index=-1, unit="a.u."):
+        return self._get_energy_type("Total Local-CCSD Energy", index=index, unit=unit)
+
+
+class SerenityJob(SingleJob):
+    """A Subclass of the SingleJob class representing a computational job with Serenity."""
+
+    _result_type = SerenityResults
+
+    def _get_ready(self):
+        """If molecule is defined, xyz files are generated accordingly. If there is data to transfer between ADF and Serenity, file formats are converted."""
+        SingleJob._get_ready(self)
```

Comment:
```suggestion
        self._get_ready()
```
(as this inherits from SingleJob)